src/acquisition/covid_hosp/common/utils.py

```diff
@@ -160,7 +160,8 @@ def merge_by_key_cols(dfs, key_cols, logger=False):
       ## repeated concatenation in pandas is expensive, but (1) we don't expect
       ## batch sizes to be terribly large (7 files max) and (2) this way we can
       ## more easily capture the next iteration's updates to any new keys
-      new_rows = df.loc[[i for i in df.index.to_list() if i not in result.index.to_list()]]
+      result_index_set = set(result.index.to_list())
+      new_rows = df.loc[[i for i in df.index.to_list() if i not in result_index_set]]
```

Comment:
Minor suggestion (non-blocking): wonder if [`pandas.Index.difference`](https://devdocs.io/pandas~1/reference/api/pandas.index.difference) would help here. I dug around under the hood and it looks like Pandas removes duplicates in both indexes and then punts to the Numpy function [`in1d`](https://github.com/numpy/numpy/blob/v1.24.0/numpy/lib/arraysetops.py#L923-L965), which uses a couple other tricks. Might not be C-fast since the keys here are probably strings, so Numpy will be holding Python object pointers, but will avoid the Numpy -> Python list conversion (in to_list), and it looks pretty clean.

```py
new_rows = df.loc[[df.index.difference(result.index)]]
```

Reply:
ugh, you tell me **_NOW_** after i dismantled all the scaffolding i had to test this stuff with real data...  😂 

that looks like it would certainly do the trick and is probably just as fast (if not faster), but it does a few weird in treating the arguments as sets (sorting or throwing away ordering).  what's in this PR so far should produce equivalent results to what it replaces, and its fast compared to the DB operations that come later in the pipeline (and its extremely faster than what was there before), so im happy to leave it as-is for now.